src/main/kotlin/org/ptss/support/infrastructure/persistence/entities/InvitationEntity.kt

```diff
@@ -0,0 +1,74 @@
+package org.ptss.support.infrastructure.persistence.entities
+
+import io.quarkus.hibernate.orm.panache.kotlin.PanacheCompanion
+import java.time.OffsetDateTime
+import java.util.UUID
+import jakarta.persistence.*
+import org.ptss.support.domain.config.InvitationProperties
+import org.ptss.support.domain.constants.ValidationConstraints.EMAIL_MAX_LENGTH
+import org.ptss.support.domain.constants.ValidationConstraints.VERIFICATION_CODE_LENGTH
+import org.ptss.support.domain.enums.Role
+
+@Entity
+@Table(
+    name = "invitations",
+    uniqueConstraints = [
+        UniqueConstraint(columnNames = ["email"])
+    ]
+)
+class InvitationEntity : BaseEntity() {
+    @Column(nullable = false, updatable = false, length = EMAIL_MAX_LENGTH)
+    lateinit var email: String
+
+    @Enumerated(EnumType.STRING)
+    @Column(nullable = false, updatable = false)
+    lateinit var role: Role
+
+    @Column(nullable = false, updatable = false, length = VERIFICATION_CODE_LENGTH)
+    lateinit var verificationCode: String
+
+    @Column(nullable = false, updatable = false)
+    lateinit var groupId: UUID
+
+    @Column(nullable = false)
+    lateinit var expiresAt: OffsetDateTime
+
+    @Column(nullable = false)
+    var isVerified: Boolean = false
+
+    @Column(nullable = false)
+    var isRegistered: Boolean = false
+
+    companion object : PanacheCompanion<InvitationEntity> {
+        fun findByEmailAndVerificationCode(email: String, code: String): InvitationEntity? =
+            find("email = ?1 and verificationCode = ?2 and isVerified = false and expiresAt > CURRENT_TIMESTAMP",
+                email, code).firstResult()
+    }
+
+    @PrePersist
+    fun prePersist() {
+        // Explicit check that role is initialized
+        if (!::role.isInitialized) {
+            throw IllegalStateException("Role must be set before persisting invitation")
+        }
+
+        require(role.canBeInvited()) {
+            "Only patients and family members can be invited"
+        }
+
+        if (!::verificationCode.isInitialized) {
+            verificationCode = generateVerificationCode()
+        }
+        if (!::expiresAt.isInitialized) {
+            expiresAt = OffsetDateTime.now().plusHours(InvitationProperties.validityHours)
+        }
+    }
```

Comment:
Hier geldt eigenlijk hetzelfd als comment in GroupEntity, wil je dit erin blijven houden of ga je dit verplaatsen? Je gebruikt namelijk geen repositories en core (neem aan dat je dit later toevoegd).